RetailCoder.VBE/UI/Command/ExportAllCommand.cs

```diff
@@ -0,0 +1,77 @@
+using System.IO;
+using System.Windows.Forms;
+using NLog;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+using Rubberduck.Navigation.CodeExplorer;
+using Rubberduck.UI.CodeExplorer.Commands;
+
+namespace Rubberduck.UI.Command
+{
+    [CodeExplorerCommand]
+    public class ExportAllCommand : CommandBase 
+    {
+        private readonly IVBE _vbe;
+        private IFolderBrowserFactory _factory;
+
+        public ExportAllCommand(IVBE vbe, IFolderBrowserFactory folderBrowserFactory) : base(LogManager.GetCurrentClassLogger())
+        {
+            _vbe = vbe;
+            _factory = folderBrowserFactory;
+        }
+
+        protected override bool EvaluateCanExecute(object parameter)
+        {
+            if (parameter == null)
+            {
+                return Evaluate(_vbe.ActiveVBProject);
+            }
+            else if (parameter is CodeExplorerProjectViewModel)
+            {
+                var node = (CodeExplorerProjectViewModel)parameter;
+                return Evaluate((IVBProject)node.Declaration.Project);
+            }
+            else if (parameter is IVBProject)
+            {
+                return Evaluate((IVBProject)parameter);
+            }
+            else { return false; }
+        }
+
+        private bool Evaluate(IVBProject project)
+        {
+            return !project.IsWrappingNullReference && project.VBComponents.Count > 0;
+        }
+
+        protected override void OnExecute(object parameter)
+        {
+            IVBProject project;
+            if (parameter == null) { project = (_vbe.ActiveVBProject); }
+            else if (parameter is CodeExplorerProjectViewModel)
+            {
+                CodeExplorerProjectViewModel projectVM = (CodeExplorerProjectViewModel)parameter;
+                project = projectVM.Declaration.Project;
+            }
+            else { project = (IVBProject)parameter; } // for unit test in ExportAllCommand.cs
```

Comment:
I see how the if-else-chain is repeated, but if I extract that into a private method, then what is the suggested way that I do this?  Do I return both a Boolean and a object from the same private method, even though the object would be unused in the `CanExecute` method?  Otherwise, I'd still need a if-else chain in each *`Execute` method.